src/transformers/tokenization_utils_base.py

```diff
@@ -1530,6 +1530,15 @@ def __init__(self, **kwargs):
 
         # For backward compatibility we fallback to set model_max_length from max_len if provided
         model_max_length = kwargs.pop("model_max_length", kwargs.pop("max_len", None))
+
+        # To specify that 'max_length' does not set 'model_max_length'
+        if model_max_length is None and "max_length" in kwargs:
```

Comment:
Can this ever happen? If model_max_length is None, the default value that will be given is the value of `max_len`.